User/src/main/java/com/example/user/service/UserService.java

```diff
@@ -0,0 +1,68 @@
+package com.example.user.service;
+
+import com.example.user.Repository.UserRepository;
+import com.example.user.exceptions.ErrorSubmissionException;
+import com.example.user.model.ChangeData;
+import com.example.user.model.NewUserData;
+import com.example.user.model.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+    public User getUserById(long id){
+        User user=userRepository.existsById(id) ? userRepository.findById(id) : null;
+        if(user!=null){
+            return user;
+        }
+        else{
+            throw new ErrorSubmissionException("Not found user id");
+        }
+    }
+
+    public void registerUser(NewUserData userData){
+        if(userData.getFirstName().isEmpty() || userData.getLastName().isEmpty() || userData.getEmail().isEmpty() || userData.getPassw().isEmpty()){
+            throw new ErrorSubmissionException("Empty fields");
+        }
+        else if(userRepository.existsByEmail(userData.getEmail()) || userRepository.existsByPassw((userData.getPassw()).hashCode())){
+            throw new ErrorSubmissionException("Email or Password already exists");
+        }
+        else {
```

Comment:
Totalnie niepotrzebna jest klasa NewUserData i całe to mapowanie z tego obiektu na Usera. Możesz po prostu podawać Usera jako argument(po prostu podawać wszystkie pola usera z wyłączeniem id bo id będzie się samo generowało podczas wpisywania do bazy danych) i zapisać go w bazie danych jeśli jest git